### DISCUSSION TITLE
Splitting tests based on file suffix or best practice

### DISCUSSION THREAD

**Discussion by joaomarcelofm** [vitest-dev/vitest#5557] [Q&A] [answered]
Hey all!

I'm looking into split the tests we run with Vitest based on the file suffix, something like:

```
// Default - will run all "*.test.ts" files
pnpm test // (alias to vitest)

// Integration - will only run ".integration-test.ts
pnpm test:integration
```

Based on the [documentation](https://vitest.dev/guide/#configuring-vitest), I could achieve the second option by changing the `vitest.config.ts`, but that would also change the default runner. I think I could have a second config file dedicated for integration tests, but is it the ideal solution for this?
Any feedback on this is welcome!

**Comment by AriPerkkio** [comment, marked as answer]
I would recommend to use workspaces:

```ts
import { defineWorkspace } from 'vitest/config';

export default defineWorkspace([
    {
        test: {
            name: 'unit',
            include: ['**/*.unit.test.ts'],
        },
    },
    {
        test: {
            name: 'integration',
            include: ['**/*.integration.test.ts'],
            // More integration test related setup here...
        },
    },
]);
```

You can also create two separate configuration files and pass it as `vitest --config ./path/to/vitest.config.ts`.

- https://github.com/vitest-dev/vitest/discussions/4675
- https://stackblitz.com/fork/github/vitest-dev/vitest/tree/main/examples/workspace?initialPath=__vitest__/

**Reply by joaomarcelofm** [reply]
Thanks for the tip @AriPerkkio!

In our case, the tests are within the same monorepo app (it's a Nuxt application, and the integration tests are checking Nuxt's server APIs), so I think the split workspaces won't work. Having different configs isn't ideal, but it is also fine here.

**Reply by sheremet-va** [reply]
Workspace is just a term, you can name it `vitest.projects.ts` and it will work the same. It exists mainly for performance, and if you just need two different configs, then it's better to have a single workspace with those configs. This is how Vitest vscode extension also works, for example.
